graphql_compiler/schema_transformation/rename_schema.py

```diff
@@ -107,26 +191,150 @@ def rename_schema(
         for renamed_name, original_name in six.iteritems(reverse_name_map)
         if renamed_name != original_name
     }
 
-    # Rename query type fields
-    schema_ast = _rename_query_type_fields(schema_ast, renamings, query_type)
+    schema_ast = _rename_and_suppress_query_type_fields(schema_ast, renamings, query_type)
     return RenamedSchemaDescriptor(
         schema_ast=schema_ast,
         schema=build_ast_schema(schema_ast),
         reverse_name_map=reverse_name_map_changed_names_only,
     )
 
 
-def _rename_types(
+def _validate_renamings(
+    schema_ast: DocumentNode, renamings: Mapping[str, Optional[str]], query_type: str
+) -> None:
+    """Validate the renamings argument before attempting to rename the schema.
+
+    Check for fields with suppressed types or unions whose members were all suppressed. Also,
+    confirm renamings contains no enums, interfaces, or interface implementation suppressions
+    because that hasn't been implemented yet.
+
+    The input AST will not be modified.
+
+    Args:
+        schema_ast: represents a valid schema that does not contain extensions, input object
+                    definitions, mutations, or subscriptions, whose fields of the query type share
+                    the same name as the types they query. Not modified by this function
+        renamings: maps original type name to renamed name or None (for type suppression). Any name
+                   not in the dict will be unchanged
+        query_type: name of the query type, e.g. 'RootSchemaQuery'
+
+    Raises:
+        - CascadingSuppressionError if a type suppression would require further suppressions
+        - NotImplementedError if renamings attempts to suppress an enum, an interface, or a type
+          implementing an interface
+    """
+    _check_for_cascading_type_suppression(schema_ast, renamings, query_type)
+    _ensure_no_unsupported_suppression(schema_ast, renamings)
+
+
+def _check_for_cascading_type_suppression(
+    schema_ast: DocumentNode, renamings: Mapping[str, Optional[str]], query_type: str
+) -> None:
+    """Check for fields with suppressed types or unions whose members were all suppressed."""
+    visitor = CascadingSuppressionCheckVisitor(renamings, query_type)
+    visit(schema_ast, visitor)
+    if visitor.fields_to_suppress or visitor.union_types_to_suppress:
+        error_message_components = [
+            f"Type renamings {renamings} would require further suppressions to produce a valid "
+            f"renamed schema."
+        ]
+        if visitor.fields_to_suppress:
+            for object_type in visitor.fields_to_suppress:
+                error_message_components.append(f"Object type {object_type} contains: ")
+                error_message_components += [
+                    f"field {field} of suppressed type "
+                    f"{visitor.fields_to_suppress[object_type][field]}, "
+                    for field in visitor.fields_to_suppress[object_type]
+                ]
```

Comment:
micro nit: `error_message_components.extend(<generator here instead of list comprehension>)` avoids materializing the intermediate list of error messages.